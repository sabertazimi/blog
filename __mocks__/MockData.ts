const time = new Date(2022, 0, 1, 8, 0, 0).toLocaleString('zh-CN', {
  hour12: false,
});

const siteMetadata = {
  title: 'Title',
  author: 'Sabertaz',
  siteUrl: 'https://example.com',
  email: 'example@github.com',
  disqusUrl: 'https://example.com',
  landingTitles: ['A', 'B', 'C'],
  socialList: {
    github: 'sabertazimi',
    twitter: 'sabertazimi',
    facebook: 'sabertazimi',
    linkedin: 'sabertazimi',
    weibo: 'sabertazimi',
  },
  bookList: Array.from(Array(3).keys()).map(() => ({
    title: 'Title',
    author: 'Sabertaz',
    url: 'https://example.com',
    description: 'Description',
  })),
};

const basePosts = Array.from(Array(5).keys()).map(index => ({
  index,
  fields: { slug: `/${index + 1}BasicNotes/` },
  frontmatter: { title: `${index + 1} Basic Notes` },
  timeToRead: index + 1,
}));

const posts = basePosts.map((post, index) => ({
  ...post,
  fields: { slug: `/${index + 1}BasicNotes/`, gitTime: `${index + 1} time` },
  frontmatter: {
    title: `${index + 1} Basic Notes`,
    subtitle: 'Be a Stupid Learner',
    author: 'Sabertaz',
    date: '2018-08-08T00:00:00.000Z',
    tags: ['JavaScript', 'Frontend Development', 'Web Development'],
  },
  timeToRead: index + 1,
}));

const MockData = {
  time,
  siteMetadata,
  basePosts,
  posts,
};

export default MockData;
