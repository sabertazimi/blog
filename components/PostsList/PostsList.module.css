a.link.link.link.link.link {
  @apply ml-2 inline-block align-bottom text-3xl font-extrabold;
  @apply transform-gpu text-primary transition hover:translate-x-4 hover:text-secondary focus:translate-x-4 focus:text-secondary;
}
