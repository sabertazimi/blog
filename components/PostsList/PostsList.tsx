import Link from '@components/Link'
import { List } from '@components/Lists'
import type { PostMeta } from '@types'
import styles from './PostsList.module.css'

interface Props {
  posts: PostMeta[]
}

function PostsList({ posts }: Props): JSX.Element {
  return (
    <List
      className="mt-6"
      dataSource={posts}
      renderItem={({ slug, title }) => (
        <List.Item>
          <List.Item.Meta
            title={(
              <Link href={`/post/${slug}`} className={styles.link}>
                {title}
              </Link>
            )}
          />
        </List.Item>
      )}
    />
  )
}

export default PostsList
