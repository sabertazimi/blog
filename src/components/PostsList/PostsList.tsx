import type { PostMetaType } from '@types';
import { List } from 'antd';
import { Link } from 'gatsby';
import React from 'react';

interface Props {
  posts: PostMetaType[];
}

const PostsList = ({ posts }: Props): JSX.Element => (
  <List
    className="mt-6"
    dataSource={posts}
    renderItem={({ slug, title }) => (
      <List.Item>
        <List.Item.Meta
          title={
            <Link
              className="text-blue-400 transition text-span-lg transform-gpu hover:translate-x-4"
              to={`${slug}`}
            >
              {title}
            </Link>
          }
        />
      </List.Item>
    )}
  />
);

export default PostsList;
