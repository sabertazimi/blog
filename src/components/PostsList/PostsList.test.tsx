import MockData from '@MockData';
import React from 'react';
import { create } from 'react-test-renderer';
import PostsList from './PostsList';

describe('PostsList', () => {
  const mockPosts = MockData.posts;

  test('should render correctly (snapshot)', () => {
    const tree = create(<PostsList posts={mockPosts} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
