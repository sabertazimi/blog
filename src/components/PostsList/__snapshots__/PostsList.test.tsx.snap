// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList should render correctly (snapshot) 1`] = `
<div
  className="ant-list ant-list-split mt-6"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <ul
        className="ant-list-items"
      >
        <li
          className="ant-list-item"
        >
          <div
            className="ant-list-item-meta"
          >
            <div
              className="ant-list-item-meta-content"
            >
              <h4
                className="ant-list-item-meta-title"
              >
                <a
                  className="text-blue-400 transition text-span-lg transform-gpu hover:translate-x-4"
                  href="/1BasicNotes/"
                >
                  1 Basic Notes
                </a>
              </h4>
            </div>
          </div>
        </li>
        <li
          className="ant-list-item"
        >
          <div
            className="ant-list-item-meta"
          >
            <div
              className="ant-list-item-meta-content"
            >
              <h4
                className="ant-list-item-meta-title"
              >
                <a
                  className="text-blue-400 transition text-span-lg transform-gpu hover:translate-x-4"
                  href="/2BasicNotes/"
                >
                  2 Basic Notes
                </a>
              </h4>
            </div>
          </div>
        </li>
        <li
          className="ant-list-item"
        >
          <div
            className="ant-list-item-meta"
          >
            <div
              className="ant-list-item-meta-content"
            >
              <h4
                className="ant-list-item-meta-title"
              >
                <a
                  className="text-blue-400 transition text-span-lg transform-gpu hover:translate-x-4"
                  href="/3BasicNotes/"
                >
                  3 Basic Notes
                </a>
              </h4>
            </div>
          </div>
        </li>
        <li
          className="ant-list-item"
        >
          <div
            className="ant-list-item-meta"
          >
            <div
              className="ant-list-item-meta-content"
            >
              <h4
                className="ant-list-item-meta-title"
              >
                <a
                  className="text-blue-400 transition text-span-lg transform-gpu hover:translate-x-4"
                  href="/4BasicNotes/"
                >
                  4 Basic Notes
                </a>
              </h4>
            </div>
          </div>
        </li>
        <li
          className="ant-list-item"
        >
          <div
            className="ant-list-item-meta"
          >
            <div
              className="ant-list-item-meta-content"
            >
              <h4
                className="ant-list-item-meta-title"
              >
                <a
                  className="text-blue-400 transition text-span-lg transform-gpu hover:translate-x-4"
                  href="/5BasicNotes/"
                >
                  5 Basic Notes
                </a>
              </h4>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
